### DISCUSSION TITLE
How to insert a list of dict into a table using python

### DISCUSSION THREAD

**Discussion by duyvanda** [duckdb/duckdb#9175] [Q&A] [answered]
I was not able to insert a list of dict into a table using the follow code:
duckdb.sql("CREATE TABLE my_table6 AS SELECT [{'x': 1, 'y': 2, 'z': 3}, {'x': 2, 'y': 5, 'z': 5}, {'x': 2, 'y': 6, 'z': 6}] as A ")
lst_dict = [{'x': 2, 'y': 5, 'z': 5},{'x': 2, 'y': 5, 'z': 5},{'x': 10, 'y': 10, 'z': 10}]
duckdb.execute("""insert into my_table6(A) values(?)""", lst_dict)

ERROR:
InvalidInputException                     Traceback (most recent call last)
[d:\ipython_file\GE\input.ipynb](file:///D:/ipython_file/GE/input.ipynb) Cell 13 line 1
----> [1](vscode-notebook-cell:/d%3A/ipython_file/GE/input.ipynb#Y130sZmlsZQ%3D%3D?line=0) duckdb.execute("""insert into my_table6(A) values(?)""", lst_dict)

InvalidInputException: Invalid Input Error: Prepared statement needs 1 parameters, 3 given

![image](https://github.com/duckdb/duckdb/assets/68499531/e621993e-0aab-44e5-b18c-4e42abec15aa)

**Comment by eaubin** [comment, marked as answer]
There are some examples in the [python dbapi docs](https://duckdb.org/docs/archive/0.9.0/api/python/dbapi). The parameters should be part of a tuple or list:

```
duckdb.execute("""insert into my_table6(A) values(?)""", (lst_dict,))
```

**Reply by duyvanda** [reply]
Thanks mate, you are my hero !!!
